paper/paper.md

```diff
@@ -0,0 +1,175 @@
+can collect the assignments with a single command and use the auto-grading
+functionality in the normal way.
+
+Since its conception in September 2014, nbgrader has been used in a number of
+educational contexts, including courses at UC Berkeley, Cal Poly, University of
+Pretoria, University of Edinburgh, Northeastern University, Central Connecticut
+State University, KTH Royal Institute of Technology, CU Boulder, University of
+Amsterdam, George Washington University, Texas A&M, Bryn Mawr College, and
+University of Maryland; and, as of May 2018, over 10,000 nbgrader-based
+notebooks exist on GitHub. In addition to its core functionality, nbgrader has
+expanded to support a number of other features, including the ability to handle
+multiple courses on the same JupyterHub instance; the option to either include
+or hide automatically graded tests; customizable late penalties; and support for
+importing assignment files downloaded from a Learning Management System (LMS).
+
+# Statement of Need
+
+The use of computational methods has become increasingly widespread in fields
+outside of computer science [@wing2008computational]. As these disciplines begin
+to require computational tools, undergraduate curricula also begin to include
+topics in programming and computer science. However, perhaps because students
+are focused on the discipline that is the object of their study—and programming
+is likely a secondary interest—it has been shown that teaching computer science
+can be more effective when courses include interdisciplinary motivations
+[@Cortina2007; @Forte2005; @Guzdial2005]. One approach for teaching programming
+in a way that facilitates exploration with interdisciplinary questions is to
+teach students computational concepts in an interactive environment where it is
+possible to quickly write, test, and tweak small units of code. Many such
+environments exist, including Mathematica [@mathematica], Maple [@maple], MATLAB
+[@matlab], Sage [@sage] and IPython [@PerezGranger2007].
+
+In recent years, the IPython project introduced the *Jupyter Notebook*
+[@kluyver2016jupyter], an interface that is conducive to interactive and
+literate computing, where programmers can interleave prose with code, equations,
+figures, and other media. The Jupyter Notebook is ideal for educators because it
+allows them to create assignments which include instructions along with code or
+Markdown cells, in which students can provide solutions to exercises. Students
+can, for example, write code both to compute and visualize a particular result.
+Because the notebook is interactive, students can iterate on a coding problem
+without needing to switch back and forth between a command line and a text
+editor, and they can rapidly see results alongside the code which produced them.
+
+Instructors in many fields have begun using the Jupyter Notebook as a teaching
+platform. The notebook has appeared in over 100 classes [@Hamrick2016;
```

Comment:
I'd love if somehow there would be room to write more about the compmodels deployment and how nbgrader ended up enabling scaling out the course in ways not able to previously (due to size), like you did in https://developer.rackspace.com/blog/deploying-jupyterhub-for-education/

Reply:
I added a sentence about this to the final paragraph, what do you think? Unfortunately there's not much room to add a lot more; we're already over the 1000 word limit now (though there's probably a few places to make things a bit more concise).